test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTests.cs

```diff
@@ -566,12 +566,12 @@ public class HubEndpointTests
                 // This invocation should be completely synchronous
                 await client.SendInvocationAsync(methodName, nonBlocking: true).OrTimeout();
 
-                // Nothing should have been written
-                Assert.False(client.Application.Reader.TryRead(out var buffer));
-
                 // kill the connection
                 client.Dispose();
 
+                // Nothing should have been written
+                Assert.False(client.Application.Reader.TryRead(out var buffer));
```

Comment:
Why this change?

Reply:
It was never actually asserting that nothing was sent.

Reply:
As a test, I commented out https://github.com/aspnet/SignalR/blob/1a842e92eccd1a54104b0e723abcc505dd20201b/src/Microsoft.AspNetCore.SignalR.Core/HubEndPoint.cs#L361 and the test still passed despite the fact that we now erroneously sent a Completion Message. This check makes it fail (as it should) when we send a response to the non-blocking invocation.